deferred-iterator.js

```diff
@@ -2,39 +2,84 @@
 
 const AbstractIterator = require('abstract-leveldown').AbstractIterator
 const inherits = require('inherits')
+const getCallback = require('./util').getCallback
+
+const kOptions = Symbol('options')
+const kIterator = Symbol('iterator')
+const kOperations = Symbol('operations')
+const kPromise = Symbol('promise')
 
 function DeferredIterator (db, options) {
   AbstractIterator.call(this, db)
 
-  this._options = options
-  this._iterator = null
-  this._operations = []
+  this[kOptions] = options
+  this[kIterator] = null
+  this[kOperations] = []
 }
 
 inherits(DeferredIterator, AbstractIterator)
 
 DeferredIterator.prototype.setDb = function (db) {
-  const it = this._iterator = db.iterator(this._options)
+  this[kIterator] = db.iterator(this[kOptions])
 
-  for (const op of this._operations) {
-    it[op.method](...op.args)
+  for (const op of this[kOperations].splice(0, this[kOperations].length)) {
```

Comment:
Why copy this array?

Reply:
I'm using splice here to empty it out. So that we can do a sanity check immediately after:

https://github.com/Level/deferred-leveldown/blob/c197496adf54b7b44bb74838dcc342f377135ece/deferred-leveldown.js#L60-L63

Reply:
Ah! I totally read over that. Could we make this clearer like this?

```js
const operations = this[kOperations]
this[kOperations] = []
```